include/engine/routing_algorithms/routing_base_mld.hpp

```diff
@@ -281,14 +301,31 @@ void relaxOutgoingEdges(const DataFacade<Algorithm> &facade,
 
     const auto level = getNodeQueryLevel(partition, heapNode.node, args...);
 
+    static constexpr auto IS_MAP_MATCHING =
+        std::is_same_v<typename SearchEngineData<mld::Algorithm>::MapMatchingQueryHeap, Heap>;
```

Comment:
I added new `MapMatchingQueryHeap` to be able to track distance in each heap entry.